src/workerd/jsg/modules.h

```diff
@@ -198,17 +198,25 @@ class ModuleRegistryImpl final: public ModuleRegistry {
     entries.insert(Entry(specifier, kj::fwd<ModuleInfo>(info)));
   }
 
-  kj::Maybe<ModuleInfo&> resolve(const kj::Path& specifier) override {
+  void addOnDemand(const kj::Path& specifier, kj::ArrayPtr<const char> sourceCode) {
```

Comment:
`addOnDemand(...)` seems like an odd name for this. I would expect something more descriptive like `addBuiltinModule(...)`

Also, where and when would this be called? And by whom? I know this is part of a large piece of work but expanding the comment here to provide some additional detail would help me as a reviewer evaluate the correctness of this.

Reply:
I have created a demo in EW PR 4479. Will rename the function shortly.